rosidl_generator_cpp/CMakeLists.txt

```diff
@@ -45,32 +43,65 @@ if(BUILD_TESTING)
     add_compile_options(-Wall -Wextra -Wpedantic)
   endif()
 
-  include_directories(include
-    ${rosidl_generator_c_INCLUDE_DIRS})
-
   ament_add_gtest(test_bounded_vector test/test_bounded_vector.cpp)
+  if(TARGET test_bounded_vector)
+    add_dependencies(test_bounded_vector ${PROJECT_NAME})
+    target_include_directories(test_bounded_vector PUBLIC
+      ${CMAKE_CURRENT_BINARY_DIR}/${PROJECT_NAME})
+    ament_target_dependencies(test_bounded_vector
+      rosidl_runtime_cpp
+      rosidl_runtime_c)
+  endif()
```

Comment:
Why are all these extra lines for each test added? There weren't present before...

Reply:
This was done before with the `include_directories` because all headers were located in the generator packages, I can include this with `target_include_directories`, but they are ROS 2 dependencies so I think `ament_target_dependencies` is the right choice.